oci/archive/oci_dest.go

```diff
@@ -0,0 +1,137 @@
+package archive
+
+import (
+	"io"
+	"os"
+
+	"github.com/containers/image/types"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/pkg/errors"
+)
+
+type ociArchiveImageDestination struct {
+	ref          ociArchiveReference
+	unpackedDest types.ImageDestination
+}
+
+// newImageDestination returns an ImageDestination for writing to an existing directory.
+func newImageDestination(ref ociArchiveReference, ctx *types.SystemContext) (types.ImageDestination, error) {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
```

Comment:
Does this work with
```go
ref := ParseReference(…)
// possibly repeated:
dest := ref.NewImageDestination()
…
dest.Close() // deletes ref.ociRef.tempDirectory ?!
// possibly repeated:
src := ref.NewImageSource() // tempDirectory unavailable
…
src.Close() // deletes ref.ociRef.tempDirectory ?!
```

the `layout.Reference`, and `tempDirectory`, AFAICS needs to be a property of the …Source / …Destination, not of the …Reference, to match the temporary directory lifetimes which terminate at *`.Close`.